library/net.c

```diff
@@ -1017,6 +1017,10 @@ int dnet_state_move_to_dht(struct dnet_net_state *st, struct dnet_addr *addrs, i
 	err = dnet_copy_addrs_nolock(st, addrs, addrs_count);
 
 	if (!err) {
+		list_for_each_entry(other, &st->node_entry, node_entry) {
+			atomic_inc(&st->n->dht_state_num);
```

Comment:
Why do you increase number of states by, well, number of connected states while you are actually adding only single state?

This whole math looks really bad in the patch